### PR TITLE
[QF-4338] Make Surah Info Header Sticky

### PR DESCRIPTION
**Pull request by al-imam**
## Summary

Fixes the surah info modal header not staying fixed while scrolling on desktop/tablet views.
Transforms the modal into a bottom sheet/drawer style on desktop (tablet breakpoint and up) with
proper sticky header behavior and scrollable content area. Mobile view remains unchanged as it

Closes: [QF-4338](https://quranfoundation.atlassian.net/browse/QF-4338)

## Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [x] ♻️ Refactoring (no functional changes)

## Test Plan

- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [x] Manual testing performed

**Testing steps:**
1. Navigate to any surah page (`/2`)
2. Click the info icon next to the surah name to open the surah info modal
3. On desktop/tablet view (tablet breakpoint and up):
   - Verify the modal appears as a bottom sheet/drawer from the bottom
   - Scroll down within the modal content
   - Verify the header stays fixed at the top while scrolling
   - Verify the content scrolls smoothly within the modal
4. On mobile view:
   - Verify existing behavior remains unchanged (header was already fixed)
   - Verify modal still functions correctly

### Code Quality

- [x] I have performed a **self-review** of my code (file by file)
- [x] My code follows the [project style guidelines](/.github/copilot-instructions.md)
- [x] No `any` types used (or justified if unavoidable)
- [x] No unused code, imports, or dead code included
- [x] Complex logic has inline comments explaining "why"
- [x] Functions are under 30 lines and follow single responsibility

### Testing & Validation

- [x] All tests pass locally (`yarn test`)
- [x] Linting passes (`yarn lint`)
- [x] Build succeeds (`yarn build`)
- [x] Edge cases and error scenarios are handled

## Screenshots/Videos

| Before | After |
| ------ | ----- |
| image | image |

## Related PRs
https://github.com/quran/quran.com-frontend-next/pull/2726
https://github.com/quran/quran.com-frontend-next/pull/2724

## AI Assistance Disclosure

- [ ] AI tools were NOT used for this PR
- [x] AI tools were used, and I have **thoroughly reviewed and validated** all generated code

[QF-4339]: https://quranfoundation.atlassian.net/browse/QF-4339?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[QF-4338]: https://quranfoundation.atlassian.net/browse/QF-4338?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/chapters/Info/SurahInfoModal.module.scss [hunk 1]
```diff
@@ -108,3 +112,59 @@ $container-min-height: 200px;
...
+}
+
+div {
+  &.surahInfoModalContent {
+    @include breakpoints.tablet {
+      flex: 1;
+      min-block-size: 0;
+      overflow-y: auto;
+      padding-block-end: calc(1.5 * var(--spacing-mega));
+    }
```
[the remark this answers is not in the document]
Comment (al-imam) [reply src/components/chapters/Info/SurahInfoModal.module.scss:156]: Using div to increase specificity to override base modal styles without !important.

File: src/components/chapters/Info/SurahInfoModal.module.scss [hunk 2]
```diff
@@ -108,3 +112,59 @@ $container-min-height: 200px;
...
+  }
+}
+
+.bottomSheetOnDesktopHeaderSeparator {
+  @include breakpoints.tablet {
+    padding-block-start: var(--spacing-medium3-px);
+    margin-block-end: var(--spacing-micro);
+    border-block-end: var(--spacing-hairline-px) solid var(--color-borders-hairline);
+  }
+}
```
[the remark this answers is not in the document]
Comment (al-imam) [reply src/components/chapters/Info/SurahInfoModal.module.scss:168]: fixed

File: src/components/chapters/Info/SurahInfoContent.tsx [hunk 3]
```diff
@@ -23,40 +23,45 @@ const SurahInfoContent: React.FC<SurahInfoContentProps> = ({ chapterId, chapterI
...
-        className={styles.descriptionContainer}
-        // eslint-disable-next-line react/no-danger
-        dangerouslySetInnerHTML={{ __html: chapterInfo.text }}
-      />
+        <div
+          className={styles.descriptionContainer}
+          // eslint-disable-next-line react/no-danger
+          dangerouslySetInnerHTML={{ __html: chapterInfo.text }}
+        />
+      </div>
```
[the remark this answers is not in the document]
Comment (al-imam) [reply src/components/chapters/Info/SurahInfoContent.tsx:64]: Added
